dapp/src/components/TimeTravelWidget.tsx

```diff
@@ -93,20 +95,43 @@ export default function TimeTravelWidget({
     [history, baseUrl]
   );
 
+  const debouncedBlockChange = useMemo(
+    () => _.debounce(handleBlockChange, 300),
+    [handleBlockChange]
+  );
+
+  const onBlockChange = useCallback(
+    (block: string) => {
+      setCurrentBlock(block);
```

Comment:
Thank you for the PR! 
However, you stumbled upon a problem I found no way around, which is caused by this line: when `currentBlock` changes, the `bbnQuery` fires on every single change, which means if you slide around a a bit you can potentially make thousands of queries, which is great for query fees but really not ideal, whereas if you only leave the debounced block, the UI lags.
I don't have much experience with React Hooks, so I wouldn't know how to work around it.

Reply:
I think a solution might be to separate the block number being displayed from the block number that's actually used in the query, and only update the query number when `handleBlockChange` is run.
Sidenote, is there a way to easily tell when the client is sending a query?

Reply:
Wait, nevermind, I think I see the issue, the date doesn't update until the query is actually made.
Unless there's some kind of way to write a graph query which could basically let you map block numbers to dates using number ranges, I don't think there's much way around it.